### PR TITLE
Changed type on number of PushNotificationObject

### PR DESCRIPTION
**Pull request by jonathanhughes**
Changed type on number of PushNotificationObject from string to number in order to prevent an error when run on IOS.
Tested in a project that uses react-native-push-notification.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/zo0r/react-native-push-notification
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by PaitoAnderson** [comment]
If you don't mind me asking, what was the error?

From their example this is a strange one, they show a number and say "specified as string" lol
`number: 10, // (optional) Valid 32 bit integer specified as string. default: none (Cannot be zero)`

**Comment by Li357** [comment]
Looks like the package inconsistently handles `number`. In one place (scheduling local notifications) it parses it before usage:


https://github.com/zo0r/react-native-push-notification/blob/3d769ad2f661432dbeaed27f031ab13bd07985d4/index.js#L256

But in others (presenting a local nofitication immediately), it delegates to `@react-native-community/push-notification-ios` before parsing:

https://github.com/zo0r/react-native-push-notification/blob/3d769ad2f661432dbeaed27f031ab13bd07985d4/index.js#L183

And it seems like that package expects a number (based on Flow typing):

https://github.com/react-native-community/push-notification-ios/blob/e3375bffa5e79e27a5c06847b06bafd2ee540c07/js/index.js#L166

**Comment by PaitoAnderson** [comment]
@Li357 nice job digging into this! I'd approve `string | number` then.

File: types/react-native-push-notification/index.d.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export class PushNotificationObject {
     userInfo?: any;
     playSound?: boolean;
     soundName?: string;
-    number?: string;
+    number?: number;
```
Comment (PaitoAnderson) [inline types/react-native-push-notification/index.d.ts:80]:
```suggestion
    number?: string | number;
```

**Comment by jonathanhughes** [comment]
Thanks for digging into this.  For reference, here's a screenshot of the error where it tries to cast NSString to NSNumber.  
image

**Review by Li357 — APPROVED** [review]

**Comment by Li357** [comment]
LGTM. Interestingly, the native code in `@react-native-community/push-notification-ios` _can_ convert a string representation of a number to a number without parsing in JS first via `RCTConvert`. Don't know exactly why '10' fails specifically, and looks like there's an exact question from [Stack Overflow about this too](https://stackoverflow.com/questions/58975624/json-value-10-of-type-nsstring-cannot-be-converted-to-nsnumber-error-in-react).

**Review by PaitoAnderson — APPROVED** [review]

**Comment by PaitoAnderson** [comment]
Ready to merge

**Pull request merged**